scripts/work.py

```diff
@@ -1,56 +1,96 @@
-    s3_resource.Object(args.s3_reporter_bucket, key).download_file(object_id)
-    delete_array.append( { 'Key': key } ) 	   
+def download(keys_array):
+    # this obviously isn't gonna really work... we'll
+    # need to maintain the S3 path as a local filesystem
+    # path to preserve everything... or preserve them in
+    # an object of k,v
+    client = boto3.client('s3')
+    response = client.list_objects_v2(Bucket=args.s3_reporter_bucket)
 
-# run our java thingy: the Docker container workdir will have already put us
-#   in the right place to, ummm, do work
+    delete_array = []
+    s3_resource = boto3.resource('s3')
+    for key in keys_array:
+        object_id = key.rsplit('/', 1)[-1]
+        print('[INFO] downloading ' + object_id + ' from s3')
 
-# TODO: error handling?
-print '[INFO] running conversion process'
-call('datastore-histogram-tile-writer -f flatbuffer_file -o orc_file ./*')
+        s3_resource.Object(args.s3_reporter_bucket, key).download_file(object_id)
+        delete_array.append( { 'Key': key } )
+
+    # TODO: untested
+    return delete_array
+
+if __name__ == "__main__":
+    # build args
+    parser = argparse.ArgumentParser()
+    parser.add_argument('s3_reporter_bucket', type=str,help='Bucket (e.g. reporter-work-prod) in which the data we wish to process is located')
+    parser.add_argument('s3_datastore_bucket', type=str, help='Bucket (e.g. datastore-output-prod) into which we will place transformed data')
+    parser.add_argument('s3_reporter_keys', type=str, help='S3 object keys which we will operate on, found in the s3_reporter_bucket')
+    parser.add_argument('time_bucket', type=int, help='The time bucket')
+    parser.add_argument('tile_id', type=int, help='The tile ID')
+    parser.add_argument('tile_level', type=int, help='The tile level')
+    parser.add_argument('tile_index', type=int, help='The tile index')
+    args = parser.parse_args()
 
-# TODO: upload the result to s3_datastore_bucket
-s3_client = boto3.client('s3')
-for upload_file in os.listdir('.'):
-    response = s3_client.put_object(
-                Bucket = args.s3_datastore_bucket,
-                Key = upload_file,
-                ContentType = 'binary/octet-stream'
-                )
+    print('[INFO] reporter intput bucket: ' + args.s3_reporter_bucket)
+    print('[INFO] datastore output bucket: ' + args.s3_datastore_bucket)
+    print('[INFO] time bucket: ' + str(args.time_bucket))
+    print('[INFO] tile id: ' + str(args.tile_id))
+    print('[INFO] tile level: ' + str(args.tile_level))
+    print('[INFO] tile index: ' + str(args.tile_index))
 
-# delete the original keys from the s3_reporter_bucket
-print '[INFO] deleting source objects from bucket ' + args.s3_reporter_bucket
-response = s3_client.delete_objects(
-    Bucket = args.s3_reporter_bucket,
-    Delete = { 'Objects': delete_array }
-    )
+    # do work
+    delete_list = download(args.s3_reporter_keys.split(','))
+    convert()
```

Comment:
instead of using `args` as a global can you pass the bits from args into convert that it needs to use